Modix/Modules/UserInfoModule.cs

```diff
@@ -9,158 +9,170 @@
-            if (!(user is null))
+            builder.Append(FormatTimeAgo("Created", member.CreatedAt));
+
+            if (member.JoinedAt is DateTimeOffset joinedAt)
             {
-                await GetUserInfo(user);
-                return;
+                builder.Append(FormatTimeAgo("joined", joinedAt));
             }
 
-            // TODO: Check the database to see if we have anything about the user in our database.
-            await ReplyAsync("User not found.");
-        }
-
-        private static string BuildList<T>(T[] array, Func<T, string> mapper)
-        {
-            var builder = new StringBuilder();
-            for (int i = 0; i < array.Length; i++)
+            if (member.RoleIds.Count > 0)
             {
-                builder.Append(GetListSeparator(i, array.Length) + mapper(array[i]));
+                var roles = member.RoleIds.Select(x => member.Guild.Roles.Single(y => y.Id == x))
+                    .Where(x => x.Id != x.Guild.Id) // @everyone role always has same ID than guild
+                    .ToArray();
+
+                if (roles.Length > 0)
+                {
+                    Array.Sort(roles); // Sort by position: lowest positioned role is first
+                    Array.Reverse(roles); // Reverse the sort: highest positioned role is first
+
+                    builder.Append(roles.Length > 1 ? "Roles: " : "Role: ");
+                    builder.AppendLine(roles.Select(r => r.Mention).Humanize());
+                }
             }
 
-            return builder.ToString();
+            embedBuilder.Color = GetDominantColor(member);
+            embedBuilder.ThumbnailUrl = member.GetAvatarUrl();
+            embedBuilder.Author.IconUrl = member.GetAvatarUrl();
         }
 
-        private static string GetListSeparator(int i, int length)
+        private async Task AddInfractionsToEmbed(ulong userId, StringBuilder builder)
         {
-            bool atLastIndex = i == length - 1;
-            if (i == 0)
-            {
-                return string.Empty;
-            }
-            else if (length > 2 && atLastIndex)
-            {
-                return ", and ";
-            }
-            else if (length > 1 && atLastIndex)
+            var infractions = await ModerationService.SearchInfractionsAsync
+            (
+                new InfractionSearchCriteria
+                {
+                    GuildId = Context.Guild.Id,
+                    SubjectId = userId,
+                    IsDeleted = false
+                },
+                new[]
+                {
+                    new SortingCriteria
+                    {
+                        PropertyName = "CreateAction.Created",
+                        Direction = SortDirection.Descending
+                    }
+                }
+            );
+
+            var noticeCount = infractions.Count(x => x.Type == InfractionType.Notice);
```

Comment:
You should definitely write a new repository method for this, that just does the counts in the query. I did something like this in `PromotionCampaignRepository` where it counts up the total number of promotion comments, by sentiment, and returns an `IDictionary<PromotionSentiment, int>`.